src/core/math/vec2/cross.js

```diff
@@ -24,7 +24,7 @@ function cross (...params) {
   }
   let z = a[0] * b[1] - a[1] * b[0]
   out[0] = out[1] = 0
-  out[2] = z
+  out[1] = z
```

Comment:
err no a cross product only exists in 3d space, and you want to set the third component (at index 2) to Z , not at index 1

Reply:
Understood, but a vec2 function returning a vec3 result just seems wrong. How about removing this from vec2?

Reply:
no, that is totally normal, that is the maths for cross product, not something we want to change :) there are plenty of functions returning scaler (float) values aswell, it is all fine